spring-pulsar-spring-cloud-stream-binder/src/main/java/org/springframework/pulsar/spring/cloud/stream/binder/provisioning/PulsarTopicProvisioner.java

```diff
@@ -34,74 +36,63 @@
 public class PulsarTopicProvisioner implements
 		ProvisioningProvider<ExtendedConsumerProperties<PulsarConsumerProperties>, ExtendedProducerProperties<PulsarProducerProperties>> {
 
-	// TODO: Retrieve partitions through config
+	private final PulsarAdministration pulsarAdministration;
+
+	private final PulsarBinderConfigurationProperties pulsarBinderConfigurationProperties;
+
+	public PulsarTopicProvisioner(PulsarAdministration pulsarAdministration,
+			PulsarBinderConfigurationProperties pulsarBinderConfigurationProperties) {
+		this.pulsarAdministration = pulsarAdministration;
+		this.pulsarBinderConfigurationProperties = pulsarBinderConfigurationProperties;
+	}
+
 	@Override
 	public ProducerDestination provisionProducerDestination(String name,
 			ExtendedProducerProperties<PulsarProducerProperties> properties) throws ProvisioningException {
-		PulsarTopic pulsarTopic = PulsarTopic.builder(name).numberOfPartitions(1).build();
-		return new PulsarProducerDestination(pulsarTopic.topicName(), pulsarTopic.numberOfPartitions());
+
+		int partitionCount = this.pulsarBinderConfigurationProperties.partitionCount();
+		int partitionCountOnBinding = properties.getPartitionCount(); // leverage the
```

Comment:
The comment here confused me for a sec as "core" is overloaded. I know what you mean now (SCSt core) but I think the variable name says it all. I would be in favor of dropping the comment.